lib/active_merchant/billing/gateways/commerce_hub.rb

```diff
@@ -311,20 +311,31 @@ def add_decrypted_wallet(source, payment, options)
         source[:walletType] = payment.source.to_s.upcase
       end
 
+      def add_network_token(source, payment, options)
+        source[:sourceType] = 'PaymentToken'
+        source[:tokenData] = payment.number
+        source[:tokenSource] = 'NETWORK_TOKEN'
+        source[:cryptogram] = payment.payment_cryptogram
+        source[:card] = {
+          expirationMonth: format(payment.month, :two_digits),
+          expirationYear: format(payment.year, :four_digits)
+        }
+      end
+
       def add_payment(post, payment, options = {})
         source = {}
-        case payment
-        when NetworkTokenizationCreditCard
-          add_decrypted_wallet(source, payment, options)
-        when CreditCard
-          if options[:encryption_data].present?
-            source[:sourceType] = 'PaymentCard'
-            source[:encryptionData] = options[:encryption_data]
-          else
-            add_credit_card(source, payment, options)
-          end
-        when String
+
+        if payment.is_a?(String)
           add_payment_token(source, payment, options)
+        elsif payment.mobile_wallet?
```

Comment:
Can you define this method in the `Check` model? If we end up adding that payment method type in the future we will get a no method error here if it isn't added there. I should have added it there originally when I created the `mobile_wallet?` method but only added it to credit card.

```
def mobile_wallet?
  false
end
```

Reply:
Good idea!